js/modules/k6/ws/ws_test.go

```diff
@@ -274,7 +275,38 @@ func TestSession(t *testing.T) {
 		`))
 		assert.NoError(t, err)
 	})
-	assertSessionMetricsEmitted(t, stats.GetBufferedSamples(samples), "", sr("WSBIN_URL/ws-echo"), 101, "")
+
+	serverCloseTests := []struct {
+		name     string
+		endpoint string
+	}{
+		{"server_close_ok", "/ws-echo"},
+		// Ensure we correctly handle invalid WS server
+		// implementations that close the connection prematurely
+		// without sending a close control frame first.
+		{"server_close_invalid", "/ws-close-invalid"},
+	}
+
+	for _, tc := range serverCloseTests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := common.RunString(rt, sr(fmt.Sprintf(`
+			let closed = false;
+			let res = ws.connect("WSBIN_URL%s", function(socket){
+				socket.on("open", function() {
+					socket.send("test");
+				})
+				socket.on("close", function() {
+					closed = true;
+				})
+			});
+			if (!closed) { throw new Error ("close event not fired"); }
+			`, tc.endpoint)))
+			assert.NoError(t, err)
+		})
+	}
+
+	assertSessionMetricsEmitted(t, stats.GetBufferedSamples(samples), "", sr("WSBIN_URL/ws-echo-invalid"), 101, "")
```

Comment:
I would expect that you will be checking inside the loop for each of the urls

Reply:
Ah, this was a wrong edit, this assert is actually part of the previous test. I'm not asserting any metric emissions in the `server_close_*` tests, didn't think it was necessary.

Will fix this (move this assert above).

Reply:
Fixed in 58ae5f78.